analyzer/queries/queries.go

```diff
@@ -313,16 +313,18 @@ const (
 	RuntimeEVMTokenAnalysisStale = `
     SELECT
       evm_token_analysis.token_address,
-      evm_token_analysis.last_mutate_round,
       evm_token_analysis.last_download_round,
       evm_tokens.token_type,
       address_preimages.context_identifier,
       address_preimages.context_version,
-      address_preimages.address_data
+      address_preimages.address_data,
+      processed_blocks.height AS download_round
     FROM chain.evm_token_analysis
     LEFT JOIN chain.evm_tokens USING (runtime, token_address)
     LEFT JOIN chain.address_preimages ON
       address_preimages.address = evm_token_analysis.token_address
+    LEFT JOIN chain.processed_blocks ON
```

Comment:
Won't this join against all already-processed heights? Causing each token to be returned 1e7 times when we're fully synced :)
You want `SELECT ..., (SELECT max(height) FROM chain.processed_blocks WHERE chain.processed_blocksanalyzer=evm_token_analysis.runtime) AS download_round, ...`, probably.

BTW we're lacking a useful index for effective `SELECT max(height)`, you should add it. Right now, we only run that query when an analyzer intiializes, so it was fine if it was slow, but I noticed even so.

Reply:
oh shoot it's a list of all processed blocks? then yeah we want the max haha

Reply:
I think postgresql had some complaint about doing it roughly this way. although I was using `$1` instead of the `evm_token_analysis.runtime`, let me try with the latter

Reply:
ah `evm_token_analysis.runtime::text` works, to cast it to be comparable to the `processed_blocks.analyzer text` column